pkg/controller/upload_controller.go

```diff
@@ -372,12 +378,37 @@ func (c *uploadController) processUpload(req *pluginv1api.Upload) error {
 			return nil
 		} else {
 			errMsg := fmt.Sprintf("Failed to upload snapshot, %v, to durable object storage. %v", peID.String(), errors.WithStack(err))
-			_, err = c.patchUploadByStatusWithRetry(req, pluginv1api.UploadPhaseUploadError, errMsg)
-			if err != nil {
-				errMsg = fmt.Sprintf("%v. %v", errMsg, errors.WithStack(err))
+			uploadCRRetryMaximum := utils.GetUploadCRRetryMaximum(nil)
+			log.Infof("UploadCRRetryMaximum is %d", uploadCRRetryMaximum)
+			if req.Status.RetryCount < int32(uploadCRRetryMaximum) {
+				_, err = c.patchUploadByStatusWithRetry(req, pluginv1api.UploadPhaseUploadError, errMsg)
+				if err != nil {
+					errMsg = fmt.Sprintf("%v. %v", errMsg, errors.WithStack(err))
+				}
+				log.Error(errMsg)
+				return errors.New(errMsg)
+			} else {
```

Comment:
Try avoiding multiple if-else blocks. There are return statements in if block already.